tracer/test/test-applications/integrations/Samples.NetActivitySdk/Program.cs

```diff
@@ -171,6 +174,40 @@ private static void RunNonW3CId()
         }
     }
 
+    private static void RunManuallyUpdatedStartTime()
+    {
+        using (var parent = new Activity("TimeParent"))
+        {
+            parent.SetIdFormat(ActivityIdFormat.Hierarchical);
+            parent.SetStartTime(new DateTime(1980, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            parent.Start();
+            using (var manuallySetStartTime = new Activity("TimeTrigger"))
+            {
+                manuallySetStartTime.SetIdFormat(ActivityIdFormat.Hierarchical);
+                manuallySetStartTime.Start();
+                // ISSUE: Activity's can have their start time modified after they are started
+                //        We use the Activity.Parent.StartTime as a basis to check whether to 
+                //        nest an Activity as a child node. 
```

Comment:
Not suggesting we should fix it here, but this still feels inherently fragile to me. Is there anything we can/should do here to track the "original" IDs so we can determine this for sure instead of needing to use this heuristic? 🤔

Reply:
> Is there anything we can/should do here to track the "original" IDs so we can determine this for sure instead of needing to use this heuristic

Specifically, this bit of code is just making sure that an Activity that didn't get mapped to a parent Datadog context or that it didn't find the mapping to a Datadog context (I think for example if it was a root Activity) that we'll place it under the currently active trace.

So I'm not really sure what we can do.

It could honestly probably use more testing around it as it seems that the Hierarchical IDs won't work with this bit of code.